rundeckapp/grails-app/controllers/rundeck/controllers/ApiController.groovy

```diff
@@ -206,59 +209,62 @@ class ApiController extends ControllerBase{
         forward(uri: servletPath.replace('/*', "/$name"))
     }
 
-    /**
-     * API endpoints
-     */
-    /**
-     * Return true if grails configuration allows given feature, or '*' features
-     * @param name
-     * @return
-     */
-    private boolean featurePresent(def name){
-        boolean featureStatus = configurationService.getBoolean("feature.incubator.${name}", false)
 
-        def splat=configurationService.getBoolean("feature.incubator.*", false)
-        return splat || featureStatus
-    }
-    /**
-     * Set an incubator feature toggle on or off
-     * @param name
-     * @param enable
-     */
-    private void toggleFeature(def name, boolean enable){
-        grailsApplication.config.feature?.incubator?.putAt(name, enable)
-    }
+
+    @Operation(
+        method = "GET",
+        summary = "Get Rundeck System Feature Status",
+        description = "Return feature's on/off status",
+        parameters = [
+            @Parameter(
+                name='featureName',
+                in = ParameterIn.PATH,
+                description = 'Feature name without the `feature.` prefix, or blank to receive list of all system features',
+                allowEmptyValue = true,
+                required = false,
+                schema = @Schema(
+                    type='string'
+                )
+            )
+        ]
+    )
+    @ApiResponse(
+        responseCode = "200",
+        description = "Map of all system features' status if no specific one specified",
+        content = @Content(
+            mediaType = "application/json",
+            schema = @Schema(implementation = Map)
+        )
+    )
+    @Tag(name = "system")
     /**
-     * Feature toggle api endpoint for development mode
+     * API endpoint to query system features' toggle status: True/False for On/Off
+     *
+     * The URL is `/feature/{featureName}` where featureName is the specific name of the feature without `rundeck.feature.` prefix and `enabled` surfix.
+     * E.g. The configuration item name of the feature to enable runner is `rundeck.feature.runner.enabled`, to query the status of this feature
+     *      the request will be `/feature/runner`. The result of this query is an JSON object { "enabled": true/false }
+     *
+     * If `featureName` parameter is absent, the query will return all system features' status as an JSON object where the key is the feature's name and value is { "enabled": true/false }
      */
-    def featureToggle={
+    @Get(
+        uri= "/feature/{featureName}",
+        produces = MediaType.APPLICATION_JSON
+    )
+    def featureQuery(@PathVariable(name = "featureName", required = false) String featureName) {
```

Comment:
Should remove `required=false`, because it produces invalid OpenAPI spec document.

```suggestion
    def featureQuery(@PathVariable(name = "featureName") String featureName) {
```